docs/extras/modules/data_connection/vectorstores/integrations/supabase.mdx

```diff
@@ -45,7 +45,7 @@ begin
   select
     id,
     content,
-    metadata,
+    (metadata || jsonb_build_object('embedding', (embedding::text)::jsonb)) as metadata,
```

Comment:
Shouldn't we be returning `embedding` directly here?

Reply:
This looks like it'll drop metadata?

Reply:
Would alleviate a lot of the downstream parsing you're doing to separate it out of the `metadata` object

Reply:
Unfortunately we can't easily get around the conversions. If we return `embedding vector(1536)` directly from the Postgres function, pgvector will output the vector as a string instead. So we would then have to parse that string on the client side, and while it looks like a JSON array of floats on first glance, I am not sure whether it is guaranteed to follow that format. So I thought it would be better to use pgvector itself to convert it into a JSON output. So the best I could do is return it separately in a jsonb object:
```sql
select
  id,
  content,
  metadata,
  jsonb_build_object('embedding', (embedding::text)::jsonb)) as embedding,
  1 - (documents.embedding <=> query_embedding) as similarity
```
This makes it a little awkward on the client, because we now have to extract `resp.embedding.embedding`, and then optionally inject it into the Document's metadata field:
```js
const { metadata } = resp;
if (searchOptions.includeEmbeddings) {
  metadata[this.embeddingKey] = resp[this.embeddingKey][this.embeddingKey];
  // or alternatively, if we return embedding without encoding into a jsonb_object:
  // metadata[this.embeddingKey] = JSON.parse(resp[this.embeddingKey]); // not 100% sure whether this works always
}
return [
  new Document({
    metadata,
    pageContent: resp.content,
  }),
  resp.similarity,
];
```
Are you sure this simplifies things?
Alternatively, I could add an explaining comment to the SQL function, something like `-- add the embedding under the "embedding" key of the metadata`. Or maybe you have a proposition for better wording?

Reply:
Do we actually want to allow returning the raw embedding to the caller here?

It seems like we have two separate things going on here - MMR and returning raw embeddings in the metadata. My preference would be to remove `includeEmbeddings` from this PR entirely (including from the filter, it seems misused there) and just have it focus on MMR.

Reply:
Thanks for persistently reviewing this so thoroughly!

Using your inputs and given the architectural decision to always return the embeddings, I was able to simplify the feature a fair amount. Please review again.

Please note that I now had to adjust some tests, because we now always return the embeddings in the metadata, with no way to keep the old behavior. Since it's just a new field it shouldn't be a breaking change though I think.
I also tested whether the new code still works with the old version of the database function (the version from the docs before this PR). Everything seems to work as expected, with the obvious differences that the MMR tests do not work, and the `metadata.embedding` is `undefined` instead of `vectors[2]` etc.